src/main/scala/devices/debug/Debug.scala

```diff
@@ -476,32 +608,32 @@ class TLDebugModuleInner(device: Device, getNComponents: () => Int, beatBytes: I
     DMSTATUSRdData.authenticated := true.B // Not implemented
     DMSTATUSRdData.version       := 2.U    // Version 0.13
 
-    when (selectedHartReg >= nComponents.U) {
-      DMSTATUSRdData.allnonexistent := true.B
-      DMSTATUSRdData.anynonexistent := true.B
-    }.elsewhen (io.debugUnavail(selectedHartReg)) {
-      DMSTATUSRdData.allunavail := true.B
-      DMSTATUSRdData.anyunavail := true.B
-    }.elsewhen (haltedBitRegs(selectedHartReg)) {
-      DMSTATUSRdData.allhalted := true.B
-      DMSTATUSRdData.anyhalted := true.B
-    }.otherwise {
-      DMSTATUSRdData.allrunning := true.B
-      DMSTATUSRdData.anyrunning := true.B
-    }
-    DMSTATUSRdData.allhavereset := haveResetBitRegs(selectedHartReg)
-    DMSTATUSRdData.anyhavereset := haveResetBitRegs(selectedHartReg)
-
     val resumereq = io.innerCtrl.fire() && io.innerCtrl.bits.resumereq
 
-    when (io.innerCtrl.fire()){
-      when (io.innerCtrl.bits.ackhavereset) {
-        haveResetBitRegs(io.innerCtrl.bits.hartsel) := false.B
+    DMSTATUSRdData.anynonexistent := (selectedHartReg >= nComponents.U)   // only hartsel can be nonexistent
+    DMSTATUSRdData.allnonexistent := (selectedHartReg >= nComponents.U) & ~hamaskFull.reduce(_ | _)
```

Comment:
can you put a parens around the `~` and the `reduce`? I'm not even quite sure the order you mean here